### PR TITLE
Feature: validator command for use with dev builds

### PR DESCRIPTION
**Pull request by jackdelahunt**
### Description
Added a command called validate that can be used when a host build with the dev tag. This command runs validate on all non-charmil commands that the host command contains.

### Type of change

- [x] New feature (non-breaking change which adds functionality)

### Checklist

- [ ] Code Review completed
- [ ] Verified independently by reviewer

## Screenshots
Validate command is ran on a dev build which prints the commands added by the host that fail validation based on the default rule config.
image

### CODE REVIEW

**Comment by jackdelahunt** [comment]
Looking for suggestions on this pr. Explicetly checking for charmil name [here](https://github.com/aerogear/charmil/pull/127/files#diff-c5c4f57f7d74bd81639f9c9ac756d9f399dc576af3eb65d02bfefa68fb2fb906R55-R57) is pretty messy plus you have to do it when running the host's children. This could be another +1 for making charmil commands their own root.

**Comment by wtrocki** [comment]
Charmil is 

- Plugin system that enables developers to build complex multi repository clis

- Validator that runs during developent time that ensures that cli meets criteria set by host cli etc.

- Distribution/Registry of clis that can be dynamically embeded (This is currently suspended until we get first two)


TLDR?
Componetize and reuse
Validate
Distrubute

We need that info in readme.

Now.. this PR doesnt fit any of those use cases as it add code to cli that will be included in end user CLI..They do not need to validate it.

Idea is also that plugins validate that during dev time so host do not need it. As mentioned before I'm not sure if we go this way it will be little bit confusing for devs and in our docs

File: core/commands/charmil.go [hunk 1]
```diff
@@ -10,13 +10,20 @@ func AttachCharmilCommands(hostRoot *cobra.Command) error {
...
 
-         // Placeholder command. To be replaced later by actual commands
+	// Placeholder command. To be replaced later by actual commands
 	cmd := &cobra.Command{
 		Use:     "charmil",
 		Short:   "built in charmil commands",
 		SilenceUsage: false,
 	}
 
+	// charmil custom commands
```
Comment (wtrocki) [inline core/commands/charmil.go:20]: Confusing - Charmil commands will be dedicated to end user.
Like for registry it will be view plugins to install or manage config.
Validator is not charmill SDK - it is separate component

**Comment by wtrocki** [comment]
> This could be another +1 for making charmil commands their own root.

That was my assumption but only for validator.

we have 2 types of commands:

- User facing 
- Development/tooling

We should not mix them as it will be uber confusing for anyone to wrap head around it.

I think this PR can be adapted easily so no worries

**Comment by wtrocki** [comment]
This brought good discusion. I think we can close this for now
